### PR TITLE
Upgrade ramen to latest snapshotter crd and enforce VGS name length limit

### PR DESCRIPTION
**Pull request by BenamarMk**
This PR includes the following updates:

1. Update Ramen to use latest CRD from external-snapshotter

- Integrated the latest CRD from external-snapshotter.
- Updated dependencies and relevant configuration files.
- Changed `VolumeSnapshotRefList` in `VolumeGroupSnapshotStatus` to `PVCVolumeSnapshotRefList`, which allows to map a `VolumeSnapshot` to the application `PVC`.

2. Ensure VolumeGroupSnapshot name contains at most 63 characters:

- Added validation to ensure that the VolumeGroupSnapshot (VGS) name complies with the Kubernetes resource name limit of 63 characters.

### CODE REVIEW

File: hack/test/groupsnapshot.storage.k8s.io_volumegroupsnapshotclasses.yaml [hunk 1]
```diff
@@ -92,4 +91,4 @@ spec:
         type: object
     served: true
     storage: true
-    subresources: {}
```
Comment (ELENAGER) [inline hack/test/groupsnapshot.storage.k8s.io_volumegroupsnapshotclasses.yaml:95]: @BenamarMk , this file is not actually changed?

File: internal/controller/cephfscg/volumegroupsourcehandler.go [hunk 2]
```diff
@@ -285,13 +292,13 @@ func (h *volumeGroupSourceHandler) RestoreVolumesFromSnapshot(
 
 	volumeSnapshot := &vsv1.VolumeSnapshot{}
 	if err := h.Client.Get(ctx,
-		types.NamespacedName{Name: vsRef.Name, Namespace: vsRef.Namespace},
+		types.NamespacedName{Name: vsName, Namespace: pvc.Namespace},
```
Comment (ELENAGER) [inline internal/controller/cephfscg/volumegroupsourcehandler.go:295]: Maybe it's better to look for volume snapshot by vs name and vs namespace? What if one day vs namespace will be different from pvc namespace?

Comment (BenamarMk) [reply internal/controller/cephfscg/volumegroupsourcehandler.go:295]: The `VolumeGroupSnapshot` status looks like this:
```
status:
  boundVolumeGroupSnapshotContentName: groupsnapcontent-66e442b9-d739-48cc-9158-a6472c3a2022
  creationTime: "2024-09-13T13:05:00Z"
  pvcVolumeSnapshotRefList:
  - persistentVolumeClaimRef:
      name: busybox-cg2-pvc1
    volumeSnapshotRef:
      name: snapshot-bc3b0f7a8da345fd3d423eebe43cf6023ad34b03d7ff1df3fdf227da330303e6-2024-09-13-1.5.0
  - persistentVolumeClaimRef:
      name: busybox-cg2-pvc2
    volumeSnapshotRef:
      name: snapshot-811a028ea7008016c74c082430686955ae178d56cc4f0fdc3e5394da2cc20a1f-2024-09-13-1.5.3
  readyToUse: true
  ```
We have only the names in the status. The snapshot is for the PVC and we already have the PVC for it.  Therefore we use the PVC namespace.  The PVC namespace is the same as the `VolumeGroupSnapshot`.

File: hack/test/groupsnapshot.storage.k8s.io_volumegroupsnapshotclasses.yaml (hunk 1 above)
Comment (BenamarMk) [reply hack/test/groupsnapshot.storage.k8s.io_volumegroupsnapshotclasses.yaml:95]: Ok, thanks. I'll remove it.

**Review by ELENAGER — APPROVED** [review]

**Review by ShyamsundarR — APPROVED** [review]

**Comment by ShyamsundarR** [comment]
LGTM, requeued the unit test run before merge.

**Pull request merged**